src/client/common/application/types.ts

```diff
@@ -1042,3 +1047,25 @@ export const IActiveResourceService = Symbol('IActiveResourceService');
 export interface IActiveResourceService {
     getActiveResource(): Resource;
 }
+
+export const ICustomEditorService = Symbol('ICustomEditorService');
+export interface ICustomEditorService {
+    /**
+     * Returns a boolean indicating if custom editors are supported or not.
+     */
+    readonly supportsCustomEditors: boolean;
+    /**
+     * Register a new provider for webview editors of a given type.
+     *
+     * @param viewType  Type of the webview editor provider.
+     * @param provider Resolves webview editors.
+     * @param options Content settings for a webview panels the provider is given.
+     *
+     * @return Disposable that unregisters the `WebviewCustomEditorProvider`.
+     */
+    registerWebviewCustomEditorProvider(viewType: string, provider: WebviewCustomEditorProvider, options?: WebviewPanelOptions): Disposable;
+    /**
+     * Opens a file with a custom editor
+     */
+    openEditor(file: Uri): Thenable<void | undefined>;
```

Comment:
Why not `Promise<void>` Why `void | undefined`?
```suggestion
    openEditor(file: Uri): Promise<void>;
```

Reply:
Because it maps internally to calling something else that returns that.

---
In reply to: [372156630](https://github.com/microsoft/vscode-python/pull/9812#discussion_r372156630) [](ancestors = 372156630)

Reply:
Actually I can just await inside. You're right, it makes more sense to return a promise.

---
In reply to: [372479925](https://github.com/microsoft/vscode-python/pull/9812#discussion_r372479925) [](ancestors = 372479925,372156630)